src/modules/include/moduleManager.hpp

```diff
@@ -11,21 +11,23 @@
 
 template<typename T>
 concept Module = requires(T t, const configuration::ConfigurationParser& configurationParser, const std::string & query,
-                             const std::shared_ptr<IMultiTypeQueue> queue) {
+                             const std::function<int(Message)>& pushMessage) {
     { t.Start() } -> std::same_as<void>;
     { t.Setup(configurationParser) } -> std::same_as<void>;
     { t.Stop() } -> std::same_as<void>;
     { t.ExecuteCommand(query) } -> std::same_as<std::string>;
     { t.Name() } -> std::same_as<std::string>;
-    { t.SetMessageQueue(queue) } -> std::same_as<void>;
+    { t.SetPushMessageFunction(pushMessage) } -> std::same_as<void>;
 };
```

Comment:
We are not using this, right? I would remove it if so, it's a problem to maintain code that it is not being used.